source/Coop/Mod/CoopServer.cs

```diff
@@ -265,25 +279,61 @@ private void ReceiveClientRequestWorldData(ConnectionBase connection, Packet pac
                     connectionServer);
         }
 
-        [GameServerPacketHandler(ECoopServerState.Preparing, EPacket.Client_DeclineWorldData)]
-        private void ReceiveClientDeclineWorldData(ConnectionBase connection, Packet packet)
+        [GameServerPacketHandler(ECoopServerState.Preparing, EPacket.Client_RequestGameData)]
+        private void SendGameData(ConnectionBase connection, Packet packet)
         {
-            m_CoopServerSMs[(ConnectionServer)connection].StateMachine.Fire(ECoopServerTrigger.DeclineWorldData);
+            PlayerHeroSerializer serializer = (PlayerHeroSerializer)CommonSerializer.Deserialize(packet.Payload);
+            Hero newHero = (Hero)serializer.Deserialize();
+            Guid newHeroGuid = CoopObjectManager.AddObject(newHero);
+
+            CoopObjectManager.AddObject(newHero);
+            CoopObjectManager.AddObject(newHero.CharacterObject);
+            CoopObjectManager.AddObject(newHero.PartyBelongedTo);
+            CoopObjectManager.AddObject(newHero.PartyBelongedTo.Party);
+            CoopObjectManager.AddObject(newHero.Clan);
+            CoopObjectManager.AddObject(newHero.PartyBelongedTo.PartyComponent.HomeSettlement);
+
+            GameData gameData = new GameData();
+            gameData.PlayerHeroId = newHeroGuid;
+
+            byte[] data = CommonSerializer.Serialize(gameData);
+
+            connection.Send(new Packet(EPacket.Server_GameData, data));
+
+            ConnectionServer connectionServer = (ConnectionServer)connection;
+            m_CoopServerSMs[connectionServer].StateMachine.Fire(ECoopServerTrigger.RequiresWorldData);
         }
 
-        [GameServerPacketHandler(ECoopServerState.SendingWorldData, EPacket.Client_Loaded)]
+        [GameServerPacketHandler(ECoopServerState.SendingGameData, EPacket.Client_Loaded)]
         private void ReceiveClientLoaded(ConnectionBase connection, Packet packet)
         {
-            m_CoopServerSMs[(ConnectionServer)connection].StateMachine.Fire(ECoopServerTrigger.ClientLoaded);
+            ConnectionServer connectionServer = (ConnectionServer)connection;
+            m_CoopServerSMs[connectionServer].StateMachine.Fire(ECoopServerTrigger.ClientLoaded);
+
+            OnServerSentWorldData?.Invoke();
         }
 
+#if DEBUG
+        [GameServerPacketHandler(ECoopServerState.SendingGameData, EPacket.BadID)]
```

Comment:
What is this for?

Reply:
Debugging if an object doesn't exist on a client or server